src/wp-includes/class-wp-user.php

```diff
@@ -513,6 +531,10 @@ public function get_role_caps() {
 
 		$wp_roles = wp_roles();
 
+		if ( ! $this->loaded_caps ) {
+			$this->caps = $this->get_caps_data();
+		}
+
```

Comment:
This is unnecessary I think. The `WP_User` class has always populated `$this->caps` before, and not doing so anymore would be a breaking change. So we don't need this here, since `$this->caps` will always be correct prior to calling this method.
```suggestion
```

Reply:
The lines below checks if caps is an array. 

```
if ( is_array( $this->caps ) ) 
```

I wanted to ensure `$this->caps` was set correctly before doing this check.

Reply:
My point is that that's unnecessary to check, because it'll always be set correctly based on `$this->caps = $this->get_caps_data()`, which is called before this `get_role_caps()` method everywhere.

Reply:
The method `get_role_caps` is public. What you create a WP_User instance and the call `get_role_caps`?  It feels safer to just make sure that caps are loaded here. What does it hurt to have this here?

Reply:
Hmm, that's a fair point. Could you add an inline comment to clarify this? For example:
```suggestion
		// Edge-case: In case someone calls this method before lazy initialization, we need to initialize on demand.
		if ( ! $this->loaded_caps ) {
			$this->caps = $this->get_caps_data();
		}

```

Reply:
In addition, ideally this method would also set `$this->loaded_caps = true` in that situation: If someone calls `get_role_caps()` before initialization, the data is all set up the same way as `load_capability_data()` does it. So there would be no point of lazy-loading that data again afterwards.